hoomd/simulation.py

```diff
@@ -240,8 +240,16 @@ def operations(self, operations):
                 self._operations._unschedule()
                 self._operations._simulation = None
                 operations._simulation = self
-                operations._schedule()
                 self._operations = operations
+                # This _schedule method must be called  after setting
+                # _operations so that self.state.update_group_dof runs without
+                # errors. This is called by the integrator when it attaches and
+                # requires that the state have access to the current integrator,
+                # so if we do not add the correct operations object to the
+                # simulation first, the new integrator will call
+                # `state.update_group_dof` but the state will still point to the
```

Comment:
Not for this PR, but we still need to implement hooks to call `update_group_dof` as needed to keep things up to date after attaching.